### PR TITLE
Refactor VQ-VAE example to work with recent versions of tensorflow-probability

### PR DESCRIPTION
**Pull request by reinvantveer**
This is a follow-up of #911 
With help from Christopher Suter, this removes the necessity for pinning the tensorflow-probability package, see https://github.com/tensorflow/probability/issues/1576 for the help he provided.

I took the liberty to rename the `x` variable to `layers`, which I believe to be more apt.

### CODE REVIEW

File: examples/generative/md/vq_vae.md [hunk 1]
```diff
@@ -795,10 +795,10 @@ them to our decoder to generate novel images.
 ```python
 # Create a mini sampler model.
 inputs = layers.Input(shape=pixel_cnn.input_shape[1:])
-x = pixel_cnn(inputs, training=False)
-dist = tfp.distributions.Categorical(logits=x)
-sampled = dist.sample()
-sampler = keras.Model(inputs, sampled)
+layers = pixel_cnn(inputs, training=False)
```
Comment (fchollet) [inline examples/generative/md/vq_vae.md:798]: This is not a layer or set of layers, but a tensor. We should name it something that reflects this, e.g. `outputs`

**Review by fchollet — COMMENTED** [review]
Thanks for the PR!

**Review by fchollet — APPROVED** [review]
Thank you!
